environment.py

```diff
@@ -85,6 +85,19 @@ def findCellNeighbors(self):
             for j in range(self.height):
                 self.grid[i][j].findNeighbors(self.neighborhoodMode)
 
+    def findCellRanges(self):
+        minDistance = min(self.sugarscape.configuration["agentVision"][0], self.sugarscape.configuration["agentMovement"][0])
+        maxDistance = min(self.sugarscape.configuration["agentVision"][1], self.sugarscape.configuration["agentMovement"][1])
```

Comment:
I sent you off on an incorrect approach. Setting a min and max before runtime won't account for diseases which inhibit (or expand) vision/movement. Perhaps we should take a _boil the ocean_ approach and go from `0...max(environmentHeight, environmentWidth)` or something similar. We compute the distance from each cell to every other cell.

I suppose this can be represented as a table in an `Environment` object. Any object (cell or agent) needing to look up distance can consult the table. That sounds like an appropriately ugly yet fast solution. Perfect for memoization.

Reply:
Finding all cells within a given range would take a constant number of list operations (grabbing each cell up to some distance away in all relevant directions based on vision/movement mode).

Reply:
We _could_ have a table in `Environment`, but since the ranges are already tied to the cells, I don't see why they shouldn't be a `Cell` variable. I added logic to account for disease penalties.